src/qibo/backends/numpy.py

```diff
@@ -25,8 +30,16 @@ def set_threads(self, nthreads):
     def cast(self, x, dtype=None, copy=False):
         if dtype is None:
             dtype = self.dtype
+        if isinstance(x, self.tensor_types):
+            return x.astype(dtype, copy=copy)
```

Comment:
I believe this case is not needed. It will just fall to the last `return np.array(x, ...)` which does the same thing.

Reply:
I also thought so, but some tests will fail if I remove it. So it is better to keep it.